Source/Core/Common/DynamicLibrary.cpp

```diff
@@ -105,7 +105,7 @@ void DynamicLibrary::Close()
 void* DynamicLibrary::GetSymbolAddress(const char* name) const
 {
 #ifdef _WIN32
-  return reinterpret_cast<void*>(GetProcAddress(reinterpret_cast<HMODULE>(m_handle), name));
+  return reinterpret_cast<void*>(GetProcAddress(static_cast<HMODULE>(m_handle), name));
 #else
   return reinterpret_cast<void*>(dlsym(m_handle, name));
```

Comment:
You missed two outer `reinterpret_cast` here. And having said that, I guess these three casts are entirely unneeded, since `HMODULE` is ultimately a typedef for `void*` and these functions effectively return `void*`, but maybe it's appropriate to leave the casts for clarity/correctness? Someone else might have a stronger opinion than me.

Reply:
This may be true but `HMODULE` is usually an alias for `HINSTANCE` which in turn is a pointer to a dummy struct called `HINSTANCE__` for type safety reasons. See Microsoft Learn's [STRICT Compliance](https://learn.microsoft.com/en-us/windows/win32/winprog/strict-compliance) document.

Reply:
Does that mean everything is fine?

Reply:
The `GetProcAddress` casts are fine. It returns a "generic" function pointer, not `void*`. Your compiler may not warn about this though. And `HMODULE` is a pointer but usually not a `void*`.

However the [dlsym(3)](https://www.man7.org/linux/man-pages/man3/dlsym.3.html) one could be removed because it already returns `void*`: